manim/mobject/geometry.py

```diff
@@ -1428,143 +1432,313 @@ def __init__(self, start_anchor, start_handle, end_handle, end_anchor, **kwargs)
         self.set_points([start_anchor, start_handle, end_handle, end_anchor])
 
 
-class Polygon(VMobject):
-    """A shape created by defining its vertices.
+class Polygram(VMobject):
+    """A generalized :class:`Polygon`, allowing for disconnected sets of edges.
 
     Parameters
     ----------
-    vertices : :class:`list`
-        The vertices of the mobject. The first one is repeated to close the shape. Must define 3-dimensions: ``[x,y,z]``
-    color : :class:`~.Colors`, optional
-        The color of the polygon.
-    kwargs : Any
-        Additional arguments to be passed to :class:`.~VMobject`
+    vertex_groups
+        The groups of vertices making up the :class:`Polygram`.
+
+        The first vertex in each group is repeated to close the shape.
+        Must define 3-dimensions: ``[x,y,z]``
+    color
+        The color of the :class:`Polygram`.
+    kwargs
+        Forwarded to the parent constructor.
 
     Examples
     --------
+    .. manim:: PolygramExample
 
-    .. manim:: PolygonExample
-        :save_last_frame:
+        import numpy as np
 
-        class PolygonExample(Scene):
+        class PolygramExample(Scene):
             def construct(self):
-                isosceles = Polygon([-5, 1.5, 0], [-2, 1.5, 0], [-3.5, -2, 0])
-                position_list = [
-                    [4, 1, 0],  # middle right
-                    [4, -2.5, 0],  # bottom right
-                    [0, -2.5, 0],  # bottom left
-                    [0, 3, 0],  # top left
-                    [2, 1, 0],  # middle
-                    [4, 3, 0],  # top right
-                ]
-                square_and_triangles = Polygon(*position_list, color=PURPLE_B)
-                self.add(isosceles, square_and_triangles)
+                hexagram = Polygram(
+                    [[0, 2, 0], [-np.sqrt(3), -1, 0], [np.sqrt(3), -1, 0]],
+                    [[-np.sqrt(3), 1, 0], [0, -2, 0], [np.sqrt(3), 1, 0]],
+                )
+                self.add(hexagram)
+
+                self.play(MoveAlongPath(Dot(), hexagram), run_time=5, rate_func=linear)
+
+
     """
 
-    def __init__(self, *vertices, color=BLUE, **kwargs):
-        VMobject.__init__(self, color=color, **kwargs)
-        # There are actually four corners, and the first one is repeated twice to form the four vertices.
-        self.set_points_as_corners([*vertices, vertices[0]])
+    def __init__(self, *vertex_groups: Sequence[Sequence[float]], color=BLUE, **kwargs):
```

Comment:
It actually can't work with `Iterable` atm because I use indexing on the vertex groups, which makes it a `Sequence`. However, I *could* make it work with any iterable but I forget why I didn't, because I did run into some problem I didn't think was worth it. Should've commented more